datajoint/table.py

```diff
@@ -72,7 +72,145 @@ def declare(self, context=None):
+                    if attr['type'].startswith('external'):
+                        continue
+
+                    #old_name and name are the same if no rename
+                    attr['old_name'] = attr['name']
+                    attr['comment'] = attr['comment'].rstrip('#')
+                    
+                    #extract rename, and new comment if necessary. brackets after any alphanumerical are ignored.
+                    rename = re.match(r"[^\w]*{[a-z][a-z0-9_]*}", attr['comment'])
+                    if rename:
+                        rename = True
+                        attr['old_name'] = re.split(r"{|}", attr['comment'])[1]
+                        attr['comment'] = re.split(r"{|}", attr['comment'])[2]
+
+                    attr['comment'] = attr['comment'].replace('"', '\\"')   # escape double quotes in comment
+
+                    #Extract changes in default, and nullable
+                    if 'default' not in attr:
+                        attr['default'] = ''
+                    attr = {k: v.strip() for k, v in attr.items()}
+                    attr['nullable'] = attr['default'].lower() == 'null'
+                    literals = ['CURRENT_TIMESTAMP'] 
+                    if attr['nullable']:
+                        if in_key:
+                            raise DataJointError('Primary key attributes cannot be nullable in line %s' % line)
+                        attr['default'] = 'NULL'
+                    else:
+                        if attr['default']:
+                            quote = attr['default'].upper() not in literals and attr['default'][0] in '"\''
+                            attr['default'] = ('"'+attr['default'][1:len(attr['default'])-1]+'"' if quote else attr['default'])
+                        else:
+                            attr['default'] = None
+
+                    #add attribute
+                    if attr['name'] not in self.heading.attributes and not rename:
+                        alter_sql += ("ADD COLUMN" 
+                                    + (' ' + attr['name'])
+                                    + (' ' + attr['type']) 
+                                    + (' ' + 'DEFAULT ' + attr['default'] if attr['default'] else '') 
+                                    + (' ' + 'COMMENT "' + attr['comment'] + '"' if attr['comment'] else '') + ', ')
+                        new_attributes.append(attr)
+                        continue
+
+                    #change attribute
+                    column_definition = (attr['type']
+                                        + (' NOT NULL' if not attr['nullable'] else '')
+                                        + (' DEFAULT ' + attr['default'] if attr['default'] else '') 
+                                        + (' COMMENT "' + attr['comment'] + '"' if attr['comment'] else ''))
+                    if (rename
+                        or self.heading.attributes[attr['old_name']].type != attr['type'] 
```

Comment:
these comparisons want to be a loop

Reply:
why would it be a loop? it is inside 
https://github.com/datajoint/datajoint-python/blob/eb6410770f9ebd154c44c4252938462b78a7c2c8/datajoint/table.py#L100
.
.
.
https://github.com/datajoint/datajoint-python/blob/eb6410770f9ebd154c44c4252938462b78a7c2c8/datajoint/table.py#L167-L168

Reply:
https://github.com/datajoint/datajoint-python/blob/c12dd9a936553bc52610b7193ab26208778f1bbf/datajoint/table.py#L168-L171